firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift

```diff
@@ -0,0 +1,221 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+
+class EditFolderViewController: UIViewController,
+                                UITableViewDelegate,
+                                UITableViewDataSource,
+                                Themeable {
+    private struct UX {
+        static let editFolderCellTopPadding: CGFloat = 25.0
+        static let parentFolderHeaderHorizzontalPadding: CGFloat = 16.0
+        static let parentFolderHeaderBottomPadding: CGFloat = 8.0
+        static let parentFolderHeaderIdentifier = "parentFolderHeaderIdentifier"
+    }
+    private enum Section: Int, CaseIterable {
+        case editFolder = 0
+        case parentFolder = 1
+    }
+    var currentWindowUUID: WindowUUID?
+    var themeManager: any ThemeManager
+    var themeObserver: (any NSObjectProtocol)?
+    var notificationCenter: any NotificationProtocol
+    var onViewWillDisappear: (() -> Void)?
+    var onViewWillAppear: (() -> Void)?
+    private var theme: any Theme {
+        return themeManager.getCurrentTheme(for: currentWindowUUID)
+    }
+
+    private let viewModel: EditFolderViewModel
+
+    private lazy var tableView: UITableView = .build { view in
+        view.dataSource = self
+        view.delegate = self
+        view.register(cellType: EditFolderCell.self)
+        view.register(cellType: OneLineTableViewCell.self)
+        view.register(UITableViewHeaderFooterView.self,
+                      forHeaderFooterViewReuseIdentifier: UX.parentFolderHeaderIdentifier)
+        view.separatorStyle = .none
+        let headerSpacerView = UIView(frame: CGRect(origin: .zero,
+                                                    size: CGSize(width: 0, height: UX.editFolderCellTopPadding)))
+        view.tableHeaderView = headerSpacerView
+    }
+
+    init(viewModel: EditFolderViewModel,
+         windowUUID: WindowUUID,
+         themeManager: any ThemeManager = AppContainer.shared.resolve(),
+         notificationCenter: any NotificationProtocol = NotificationCenter.default) {
+        self.currentWindowUUID = windowUUID
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        _ = UIImage(resource: .addToHomescreenLarge)
```

Comment:
Is this line still needed?

Reply:
No sorry, i forgot it there

Reply:
Sorry i forgot it there, i'm going to remove it.